optimize/types.go

```diff
@@ -129,11 +129,9 @@ type Problem struct {
 	Grad func(grad []float64, x []float64) []float64
 
 	// Hess evaluates the Hessian at x and stores the result in-place in hess.
-	// Hess must not modify x. Hess may use (and return) the provided Symmetric
-	// if it is non-nil, or must allocate a new Symmetric otherwise. Minimize
-	// will 'give back' the returned Symmetric where possible, allowing Hess
-	// to use a type assertion on the provided matrix.
-	Hess func(hess mat.Symmetric, x []float64) mat.Symmetric
+	// Hess must not modify x. Hess may use the provided mat.SymDense, and will
```

Comment:
In this case, `Hess` must use the provided `mat.SymDense`, and must resize it if zero-valued. With a TODO to document ReuseAs if/when it is added.

Reply:
Done the comment here, but I'm not sure what you want with the TODO.